### PR TITLE
BpkText styles inheritance support

### PR DESCRIPTION
**Pull request by MichaelTkachuk**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-react-native/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: lib/bpk-component-text/src/BpkText-test.common.js [hunk 1]
```diff
@@ -57,6 +57,34 @@ const commonTests = () => {
...
+        )
+        .toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+
+    it('should render correctly when weight is set to inherit', () => {
+      const tree = renderer
+        .create(
+          <BpkText textStyle="xxxl">
+            <BpkText weight="emphasized">Some bold text</BpkText>
```
Comment (tiagoengel) [inline lib/bpk-component-text/src/BpkText-test.common.js:78]: This should be weight="inherit" and have a weight set on the parent.

File: lib/bpk-component-text/src/BpkText.js [hunk 2]
```diff
@@ -97,6 +97,10 @@ const TEXT_TOKENS = {
 };
 
 const getStyleByTextStyle = textStyle => {
+  if (textStyle === 'inherit') {
```
Comment (tiagoengel) [inline lib/bpk-component-text/src/BpkText.js:100]: Let's remove this from here and just return an empty object in the mapping bellow.

File: lib/bpk-component-text/src/BpkText.js [hunk 3]
```diff
@@ -185,6 +193,7 @@ const styles = BpkDynamicStyleSheet.create({
   xl: { ...getStyleByTextStyle('xl') },
   xxl: { ...getStyleByTextStyle('xxl') },
   xxxl: { ...getStyleByTextStyle('xxxl') },
+  inherit: { ...getStyleByTextStyle('inherit') },
```
Comment (tiagoengel) [inline lib/bpk-component-text/src/BpkText.js:196]: Let's change this to `inherit: {}`

File: lib/bpk-component-text/src/BpkText.js [hunk 4]
```diff
@@ -141,6 +145,10 @@ const expandAndroidThemedFont = (weight, baseFontFamily) => {
 const getEmphasizeProperties = weight => {
   const emphasizeProperties = {};
 
+  if (weight === 'inherit') {
```
Comment (tiagoengel) [inline lib/bpk-component-text/src/BpkText.js:148]: Let's move this to the top of the function and just return an empty object. 

```
if (weight === 'inherit') {
  return {};
}
```

Comment (MichaelTkachuk) [reply lib/bpk-component-text/src/BpkText.js:142]: Changed

File: lib/bpk-component-text/src/BpkText.js (hunk 3 above)
Comment (MichaelTkachuk) [reply lib/bpk-component-text/src/BpkText.js]: Changed

File: lib/bpk-component-text/src/BpkText.js (hunk 2 above)
Comment (MichaelTkachuk) [reply lib/bpk-component-text/src/BpkText.js]: Changed

File: lib/bpk-component-text/src/BpkText-test.common.js (hunk 1 above)
Comment (MichaelTkachuk) [reply lib/bpk-component-text/src/BpkText-test.common.js]: Changed

File: lib/bpk-component-text/README.md [hunk 5]
```diff
@@ -41,12 +44,12 @@ export default class App extends Component {
...
-| ------------------------------------ | ---------------------------------------------------- | -------- | ------------- |
-| children                             | node                                                 | true     | -             |
-| textStyle                            | oneOf('xxl', 'xl', 'lg', 'base', 'sm', 'xs', 'caps') | false    | base          |
-| weight                               | oneOf('regular', 'emphasized', 'heavy')              | false    | regular       |
-| emphasize (deprecated, use `weight`) | bool                                                 | false    | false         |
+| Property                             | PropType                                                        | Required | Default Value |
+| ------------------------------------ | --------------------------------------------------------------- | -------- | ------------- |
+| children                             | node                                                            | true     | -             |
+| textStyle                            | oneOf('xxl', 'xl', 'lg', 'base', 'sm', 'xs', 'caps', inherit)   | false    | base          |
```
Comment (tiagoengel) [inline lib/bpk-component-text/README.md:50]: nit: 'inherit'

File: lib/bpk-component-text/README.md [hunk 6]
```diff
@@ -41,12 +44,12 @@ export default class App extends Component {
...
+| weight                               | oneOf('regular', 'emphasized', 'heavy', 'inherit')              | false    | regular       |
```
Comment (tiagoengel) [inline lib/bpk-component-text/README.md:51]: We are not supporting `inherit` for weight anymore right? We should removed this.

Comment (MichaelTkachuk) [reply lib/bpk-component-text/README.md]: Good point, fixed

File: lib/bpk-component-text/README.md [hunk 7]
```diff
@@ -48,7 +48,7 @@ export default class App extends Component {
 | ------------------------------------ | --------------------------------------------------------------- | -------- | ------------- |
 | children                             | node                                                            | true     | -             |
 | textStyle                            | oneOf('xxl', 'xl', 'lg', 'base', 'sm', 'xs', 'caps', inherit)   | false    | base          |
```
Comment (tiagoengel) [inline lib/bpk-component-text/README.md:50]:
```suggestion
| textStyle                            | oneOf('xxl', 'xl', 'lg', 'base', 'sm', 'xs', 'caps', 'inherit')   | false    | base          |
```

**Comment by tiagoengel** [comment]
Code looks good, I'll run locally and check the Android ci before approving

**Pull request merged**